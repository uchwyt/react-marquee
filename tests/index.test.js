/**
 * Created by marcin on 06.12.16.
 */
import renderer from 'react-test-renderer';
import React from 'react';
import Marquee from '../lib/index';
import { JssProvider, SheetsRegistry } from 'react-jss';

describe('ReactMarque', function () {
  const text = 'Test text';
  it('render text with default props', () => {
    'use strict';
    const tree = renderer
    .create(<Marquee text={text}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render text with custom class', () => {
    'use strict';
    const tree = renderer
    .create(<Marquee text={text} className='test-class' classInner='inner-class'/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render with loop class', () => {
    'use strict';

    const tree = renderer
    .create(<Marquee text={text} loop/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render with custom delay', () => {
    'use strict';
    const registry = new SheetsRegistry();

    const tree = renderer
    .create(
      <JssProvider registry={registry}>
        <Marquee text={text} delay={100}/>
      </JssProvider>
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
    expect(registry.toString()).toMatchSnapshot()
  });

  it('render with custom timeout', () => {
    'use strict';
    const registry = new SheetsRegistry();
    const tree = renderer
    .create(
      <JssProvider registry={registry}>
        <Marquee text={text} timeout={100}/>
      </JssProvider>
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
    expect(registry.toString()).toMatchSnapshot()
  });

  it('render with custom direction', () => {
    'use strict';
    const registry = new SheetsRegistry();
    const tree = renderer
    .create(<JssProvider registry={registry}><Marquee text={text} direction='alternate-reverse'/></JssProvider>)
    .toJSON();
    expect(tree).toMatchSnapshot();
    expect(registry.toString()).toMatchSnapshot()
  });

});
