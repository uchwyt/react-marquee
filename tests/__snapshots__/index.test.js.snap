// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactMarque render text with custom class 1`] = `
<div
  className="Marquee-root-0-1-1 Marquee-root-0-1-6 test-class"
>
  <span
    className="Marquee-inner-0-1-2 Marquee-inner-0-1-7 inner-class"
  >
    Test text
  </span>
</div>
`;

exports[`ReactMarque render text with default props 1`] = `
<div
  className="Marquee-root-0-1-1 Marquee-root-0-1-4"
>
  <span
    className="Marquee-inner-0-1-2 Marquee-inner-0-1-5"
  >
    Test text
  </span>
</div>
`;

exports[`ReactMarque render with custom delay 1`] = `
<div
  className="Marquee-root-0-1-1 Marquee-root-0-1-4"
>
  <span
    className="Marquee-inner-0-1-2 Marquee-inner-0-1-5"
  >
    Test text
  </span>
</div>
`;

exports[`ReactMarque render with custom delay 2`] = `
".Marquee-root-0-1-1 {
  overflow: hidden;
  position: relative;
  padding-left: 100%;
}
.Marquee-root-0-1-1:hover, .Marquee-root-0-1-1:hover .Marquee-inner-0-1-2 {
  animation-play-state: paused;
}
.Marquee-inner-0-1-2 {
  display: inline-block;
  white-space: nowrap;
}
@keyframes reduce {
  to {
    padding-left: 0;
  }
}
@keyframes scroll {
  to {
    transform: translateX(-100%);
  }
}
.Marquee-loop-0-1-3, .Marquee-loop-0-1-3 .Marquee-inner-0-1-2 {
  animation-iteration-count: infinite;
}
.Marquee-root-0-1-4 {
  animation: reduce 40s linear 100s 1 normal;
}
.Marquee-inner-0-1-5 {
  animation: scroll 40s linear 100s 1 normal;
}"
`;

exports[`ReactMarque render with custom direction 1`] = `
<div
  className="Marquee-root-0-1-1 Marquee-root-0-1-4"
>
  <span
    className="Marquee-inner-0-1-2 Marquee-inner-0-1-5"
  >
    Test text
  </span>
</div>
`;

exports[`ReactMarque render with custom direction 2`] = `
".Marquee-root-0-1-1 {
  overflow: hidden;
  position: relative;
  padding-left: 100%;
}
.Marquee-root-0-1-1:hover, .Marquee-root-0-1-1:hover .Marquee-inner-0-1-2 {
  animation-play-state: paused;
}
.Marquee-inner-0-1-2 {
  display: inline-block;
  white-space: nowrap;
}
@keyframes reduce {
  to {
    padding-left: 0;
  }
}
@keyframes scroll {
  to {
    transform: translateX(-100%);
  }
}
.Marquee-loop-0-1-3, .Marquee-loop-0-1-3 .Marquee-inner-0-1-2 {
  animation-iteration-count: infinite;
}
.Marquee-root-0-1-4 {
  animation: reduce 40s linear 0s 1 alternate-reverse;
}
.Marquee-inner-0-1-5 {
  animation: scroll 40s linear 0s 1 alternate-reverse;
}"
`;

exports[`ReactMarque render with custom timeout 1`] = `
<div
  className="Marquee-root-0-1-1 Marquee-root-0-1-4"
>
  <span
    className="Marquee-inner-0-1-2 Marquee-inner-0-1-5"
  >
    Test text
  </span>
</div>
`;

exports[`ReactMarque render with custom timeout 2`] = `
".Marquee-root-0-1-1 {
  overflow: hidden;
  position: relative;
  padding-left: 100%;
}
.Marquee-root-0-1-1:hover, .Marquee-root-0-1-1:hover .Marquee-inner-0-1-2 {
  animation-play-state: paused;
}
.Marquee-inner-0-1-2 {
  display: inline-block;
  white-space: nowrap;
}
@keyframes reduce {
  to {
    padding-left: 0;
  }
}
@keyframes scroll {
  to {
    transform: translateX(-100%);
  }
}
.Marquee-loop-0-1-3, .Marquee-loop-0-1-3 .Marquee-inner-0-1-2 {
  animation-iteration-count: infinite;
}
.Marquee-root-0-1-4 {
  animation: reduce 100s linear 0s 1 normal;
}
.Marquee-inner-0-1-5 {
  animation: scroll 100s linear 0s 1 normal;
}"
`;

exports[`ReactMarque render with loop class 1`] = `
<div
  className="Marquee-root-0-1-1 Marquee-root-0-1-8 Marquee-loop-0-1-3"
>
  <span
    className="Marquee-inner-0-1-2 Marquee-inner-0-1-9"
  >
    Test text
  </span>
</div>
`;
